engine/access/rest/converter.go

```diff
@@ -3,21 +3,201 @@ package rest
 import (
 	"encoding/base64"
 	"fmt"
+	"regexp"
 
 	"github.com/onflow/flow-go/engine/access/rest/generated"
 	"github.com/onflow/flow-go/model/flow"
 )
 
-// Converter provides the conversion function to/from the Swagger object to Flow objects
+// Converter provides functionality to convert from request models generated using
+// open api spec and flow models.
 
-func toBlock(flowBlock *flow.Block) *generated.Block {
+// Flow section - converting request data to flow models with validation.
+
+func toID(id string) (flow.Identifier, error) {
+	valid, _ := regexp.MatchString(`^[0-9a-fA-F]{64}$`, id)
+	if !valid {
+		return flow.Identifier{}, fmt.Errorf("invalid ID")
+	}
+
+	return flow.HexStringToIdentifier(id)
+}
+
+func toAddress(address string) (flow.Address, error) {
+	valid, _ := regexp.MatchString(`^[0-9a-fA-F]{16}$`, address)
+	if !valid {
+		return flow.Address{}, fmt.Errorf("invalid address")
+	}
+
+	return flow.HexToAddress(address), nil
+}
+
+func toProposalKey(key *generated.ProposalKey) (flow.ProposalKey, error) {
+	address, err := toAddress(key.Address)
+	if err != nil {
+		return flow.ProposalKey{}, err
+	}
+
+	return flow.ProposalKey{
+		Address:        address,
+		KeyIndex:       uint64(key.KeyIndex),
+		SequenceNumber: uint64(key.SequenceNumber),
+	}, nil
+}
+
+func toSignature(signature string) ([]byte, error) {
+	signatureBytes := []byte(signature)
```

Comment:
Is this actually correct? in the spec, the signature should be a base64 encoded string. But here, we are directly converting it to a byte array, which won't actually match the data the string is supposed to represent

Reply:
The transaction api is not yet completely implemented. It doesn't has unit tests either..it is WIP.

Reply:
Created this issue to tackle the base64 encoding - https://github.com/dapperlabs/flow-go/issues/6022